src/main/java/com/faforever/moderatorclient/ui/main_window/LadderMapPoolController.java

```diff
@@ -306,17 +346,60 @@ public void refresh() {
         selectedMap.setValue(currentMap);
     }
 
+    public List<Map> filterByFavorites(List<Map> in) {
+        List<Map> out = Lists.newArrayList();
+
+        in.forEach((map) -> {
+            if (favoritesCache.contains(Integer.parseInt(map.getId()))) {
+                out.add(map);
+            }
+        });
+        return out;
+    }
+
     public void onSearchMapVault() {
-        mapVaultView.getRoot().getChildren().clear();
-        mapVaultView.getSortOrder().clear();
-        String mapNamePattern = null;
 
-        if (filterByMapNameCheckBox.isSelected()) {
-            mapNamePattern = mapNamePatternTextField.getText();
-        }
 
-        ViewHelper.fillMapTreeView(mapVaultView,
-                mapService.findMaps(mapNamePattern).stream());
+        Task<Void> task = new Task<Void>() {
+            @Override
+            protected Void call() throws Exception {
+                String mapNamePattern = null;
+                mapNamePattern = mapNamePatternTextField.getText();
+
+                List<Map> mapsTmp;
+                String finalMapNamePattern = mapNamePattern;
+
+                if (finalMapNamePattern.equals("")) {
+                    mapsTmp = mapService.findMaps("");
+                } else {
+                    mapsTmp = mapService.findMaps('*' + mapNamePattern + '*');
+                }
+
+
+
+                List<Map> filteredMaps;
+
+                if (filterByFavorites.isSelected()) {
+                    filteredMaps = filterByFavorites(mapsTmp);
+                } else {
+                    filteredMaps = mapsTmp;
+                }
+
+
+                final List<Map> mapsFinal = filteredMaps;
+                cachedMaps.removeAll(cachedMaps.stream().toList());
+                cachedMaps.addAll(mapsFinal);
+                Platform.runLater( () -> {
+                    mapVaultView.getRoot().getChildren().clear();
+                    mapVaultView.getSortOrder().clear();
+                    ViewHelper.fillMapTreeView(mapVaultView, mapsFinal.stream());
+                });
+                return null;
+            }
+        };
+
+        new Thread(task).start();
```

Comment:
Rather than just creating threads and starting them it would be better to set up an executor service to execute them. Since this is spring boot one should even be pre configured that you just need to autowire into the class